### PR TITLE
Add libuvc

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge/help-c-cpp this is ready for review. Thanks!

File: recipes/libuvc/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS} .. \
+      -G "Ninja" \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (marcelotrevisani) [inline recipes/libuvc/build.sh:9]:
```suggestion
      -DCMAKE_PREFIX_PATH=$PREFIX \
      -DCMAKE_INSTALL_PREFIX=$PREFIX \
```

Comment (xhochy) [reply recipes/libuvc/build.sh:9]: Rather delete the whole line as these arguments are all part of CMSKE_ARGS.

**Comment by wolfv** [comment]
awesome!

File: recipes/libuvc/build.sh (hunk 1 above)
Comment (Tobias-Fischer) [reply recipes/libuvc/build.sh]: I am confused - which line(s) should go away?

File: recipes/libuvc/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS} .. \
+      -G "Ninja" \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
```
Comment (xhochy) [inline recipes/libuvc/build.sh:11]: All of these can go as they are redundant with the `CMAKE_ARGS` we pass in through e.g. https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/6290f55b73a887086b0efe0cdb32553646696a8a/recipe/activate-gcc.sh#L137-L149

```suggestion
```

File: recipes/libuvc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - ninja
+    - pkgconfig
```
Comment (xhochy) [inline recipes/libuvc/meta.yaml:24]: `pkgconfig` is the Python package, you probably only need the executable.
```suggestion
    - pkg-config
```

**Comment by Tobias-Fischer** [comment]
Looking good now, thanks for the tips @xhochy

**Pull request merged**